### PR TITLE
Fall back to load_incluster_config() when load_kube_config() fails

### PR DESCRIPTION
**Pull request by manning-ncsa**
# Description

See https://github.com/kubernetes-client/python/issues/1005 for a good description of the issue.

Fixes # (issue)

## Type of change

Choose which options apply, and delete the ones which do not apply.

- Bug fix (non-breaking change that fixes an issue)

### CODE REVIEW

**Comment by benclifford** [comment]
This is failing flake8 with:

```
flake8 parsl/
parsl/providers/kubernetes/kube.py:109:9: E722 do not use bare 'except'
```

File: parsl/providers/kubernetes/kube.py [hunk 1]
```diff
@@ -104,7 +104,10 @@ def __init__(self,
         if not _kubernetes_enabled:
             raise OptionalModuleMissing(['kubernetes'],
                                         "Kubernetes provider requires kubernetes module and config.")
-        config.load_kube_config()
+        try:
+            config.load_kube_config()
+        except:
```
Comment (benclifford) [inline parsl/providers/kubernetes/kube.py:109]: This line is caught by flake8: it catches all exceptions.

In practice in parsl, this sort of thing has caused problems such as ctrl-C handling: https://github.com/Parsl/parsl/issues/2174

Maybe it should be more narrowly `kubernetes.config.config_exception.ConfigException`

Comment (manning-ncsa) [reply parsl/providers/kubernetes/kube.py:109]: Sounds reasonable.

Comment (dannielshalev) [reply parsl/providers/kubernetes/kube.py:109]: yes it is, try this try block, and the issue persists
```
    try:
        config.load_incluster_config()
    except ConfigException as e:
        logging.exception(e)
        config.load_config()

```

**Comment by tcbonds** [comment]
@manning-ncsa You still working on this? Running into the same problem right now.

**Comment by manning-ncsa** [comment]
@tcbonds No, I am not working on this anymore. Good luck!

**Comment by mfisher87** [comment]
@benclifford myself and @trey-stafford are dealing with this today, and this looks simple enough for us to re-implement in a superceding PR so we're going to go for that :)

**Comment by benclifford** [comment]
closing because #3357 is a replacement

**Pull request closed without merging**